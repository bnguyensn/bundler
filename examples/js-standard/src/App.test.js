import React from 'react';
import {
  render,
  cleanup,
} from '@testing-library/react';
import 'jest-dom/extend-expect';
import App from './App';

afterEach(cleanup);

test.only('Application renders successfully', () => {
  const {
    getByText,
    getAllByText,
    getAllByTitle,
    getAllByAltText,
    container,
  } = render(<App />);

  // Map out our UI nodes
  const headerNode = getByText('Hello, world!');
  const sectionNodes = getAllByText("Here's a", { exact: false });
  const catImgNodes = getAllByAltText('A cat');
  const jsLogoNodes = [
    ...getAllByTitle('JavaScript logo'),
    ...getAllByAltText('JavaScript logo'),
  ];

  // Tests
  expect(container).toMatchSnapshot();
  expect(sectionNodes.length).toBe(5);
  expect(catImgNodes.length).toBe(2);
  expect(jsLogoNodes.length).toBe(3);
});
