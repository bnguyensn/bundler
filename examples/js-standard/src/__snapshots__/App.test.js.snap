// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application renders successfully 1`] = `
<div>
  <div
    class="app"
  >
    <h1>
      Hello, world!
    </h1>
    <div
      class="section"
    >
      Here's a large cat:
      <img
        alt="A cat"
        src="test-file-stub"
      />
    </div>
    <div
      class="section"
    >
      Here's a small cat (this image is inlined using url-loader):
      <img
        alt="A cat"
        src="test-file-stub"
      />
    </div>
    <div
      class="section"
    >
      Here's an SVG loaded as a React component:
      <svg-mock
        title="JavaScript logo"
      />
    </div>
    <div
      class="section"
    >
      Here's an SVG loaded as an embedded img element:
      <img
        alt="JavaScript logo"
        src="[object Object]"
        style="width: 100px; height: 100px;"
      />
    </div>
    <div
      class="section"
    >
      Here's a component with SVG background loaded using url-loader:
      <div
        class="has-svg-background"
        title="JavaScript logo"
      />
    </div>
  </div>
</div>
`;
